### PR TITLE
ECO-1043 missing fields for addresses

### PR DESCRIPTION
**Pull request by lockw1n**
- Developer(s): @aleksey-kotsuba

- Peer Reviewer:

- Ticket: https://spryker.atlassian.net/browse/ECO-1043

- Project PR: https://github.com/spryker-eco/arvato-rss

- Academy PR:

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ArvatoRss             | patch (new)           |                       |

#### Release Notes

-----------------------------------------

#### Module ArvatoRss

_**Patch:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated.
- [ ] The change is isolated and not mixed with any cleanups or other changes.
- [ ] New code fits to the architecture rules.
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer).

### Bugfixes
Additional Address field has been mapped to request and processed from response.

### CODE REVIEW

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/ArvatoRssRequestApiConfig.php [hunk 1]
```diff
@@ -32,4 +32,5 @@
     const ARVATORSS_API_PRODUCTGROUPID = 'ProductGroupId';
     const ARVATORSS_API_UNITPRICE = 'UnitPrice';
     const ARVATORSS_API_UNITCOUNT = 'UnitCount';
+    const ARVATORSS_API_STREET_ADDITIONAL = 'StreetNumberAdditional';
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/ArvatoRssRequestApiConfig.php]: Not logical name, I would include NUMBER as well in constant name

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckResponseConverter.php [hunk 2]
```diff
@@ -59,6 +59,7 @@ protected function convertAddressValidationResponse(stdClass $response)
         $addressValidationResponse->setZipCode($response->ZipCode);
         $addressValidationResponse->setCity($response->City);
         $addressValidationResponse->setCountry($response->Country);
+        $addressValidationResponse->setStreetNumberAdditional($response->StreetNumberAdditional);
```
Comment (jandert) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckResponseConverter.php]: check for existence. Field doesn't have to be there.

**Review by jandert — CHANGES_REQUESTED** [review]
Please add a test for this:
1. Response with this additional field set
2. without the additional field

**Review by profuel — CHANGES_REQUESTED** [review]
Please update functional/unit tests to cover added functionality

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/ArvatoRssRequestApiConfig.php (hunk 1 above)
Comment (jandert) [reply src/SprykerEco/Zed/ArvatoRss/Business/Api/ArvatoRssRequestApiConfig.php]: Does not have to a number. Could also say smt like "2nd floor" or so

**Review by profuel — APPROVED** [review]

**Review by jandert — APPROVED** [review]

**Pull request merged**